portfolio/src/main/webapp/script.js

```diff
@@ -31,23 +31,186 @@ function addRandomFunFact() {
         unusedFunFacts[randomIndex] = firstFunFact;
         unusedFunFacts.shift();
 
-        const funFactRowText = '<tr><td>' + funfact + '</td></tr>';
-        tableText += '<tr><td>' + funfact + '</td></tr>';
-        countTableEntries++;
+        factTableText += '<tr><td>' + funfact + '</td></tr>';
         document.getElementById('funfact-message').innerHTML = "";
         
-        funFactTable.innerHTML = tableText + '</table>';
+        funFactTable.innerHTML = factTableText + '</table>';
 
   } else {
       document.getElementById('funfact-message').innerHTML = "Congrats - you've gone through all of the fun facts!";
   }
 
 }
 
+/**
+ * Fetches the comments from the server and adds it to the DOM.
+ Based on week-3-server/random-quotes/src/webapp/script.js from the Week-3-Server tutorial
+ */
+
+var shouldDisplay = true;
+function getComments(myLocation) {
+
+  //first, reset by updating num comments, which also hides comments
+  refreshNumComments(myLocation);
+  //getNumComments(myLocation);
+  
+  //toggle between hide and display
+  if (shouldDisplay) {
+
+    console.log('Fetching comments.');
+    document.getElementById(myLocation + "-get-button").innerHTML = "Hide Comments";
+
+    setLocation(myLocation);
+
+    // The fetch() function returns a Promise because the request is asynchronous.
+    const responsePromise = fetch('/comments?location=' + myLocation, {
+        location: myLocation
+    });
+
+    // When the request is complete, pass the response into handleResponse().
+    responsePromise.then(handleResponse);
+
+    shouldDisplay = false;
+  } else {
+    hideComments(myLocation);
+  }
+}
+
+var currLocation = "";
+function setLocation(myLocation){
+    currLocation = myLocation;
+}
+
+function getLocation(){
+    return currLocation;
+}
+
+/**
+ * Handles response by converting it to text and passing the result to
+ * addCommentsToDom().
+ */
+function handleResponse(response) {
+
+  console.log('Handling the response.');
+
+  // response.text() returns a Promise, because the response is a stream of
+  // content and not a simple variable.
+  const textPromise = response.text();
+
+  // When the response is converted to text, pass the result into the
+  // addCommentsToDom() function.
+  textPromise.then(addCommentsToDom);
+}
+
+/** Adds comments to the DOM. */
+function addCommentsToDom(comments) {
```

Comment:
In L102, we call addCommentsToDom, so Im not sure how we are passing in the argument comments.

Reply:
I based this Promise/then structure on some sample code from the tutorials for Week 3. From my understanding, in _textPromise.then(addCommentsToDom);_, the Promise/then structure ensures that textPromise is passed in as the _comments_ input to addCommentsToDom.